integrations/terraform/protoc-gen-terraform-discoveryconfig.yaml

```diff
@@ -0,0 +1,79 @@
+---
+target_package_name: 'v1'
+default_package_name: 'github.com/gravitational/teleport/api/gen/proto/go/teleport/discoveryconfig/v1'
+duration_custom_type: Duration
+use_state_for_unknown_by_default: true
+
+# Top-level type names to export
+types:
+  - 'DiscoveryConfig'
+
+# These import paths were not being automatically picked up by
+# protoc-gen-terraform without these overrides
+import_path_overrides:
+  'types': 'github.com/gravitational/teleport/api/types'
+  'wrappers': 'github.com/gravitational/teleport/api/types/wrappers'
+  'durationpb': 'google.golang.org/protobuf/types/known/durationpb'
+  'timestamppb': 'google.golang.org/protobuf/types/known/timestamppb'
+  'v1': 'github.com/gravitational/teleport/api/gen/proto/go/teleport/header/v1'
+  'v11': 'github.com/gravitational/teleport/api/gen/proto/go/teleport/label/v1'
+  'github_com_gravitational_teleport_integrations_terraform_tfschema': 'github.com/gravitational/teleport/integrations/terraform/tfschema'
+
+# id field is required for integration tests. It is not used by provider.
+# We have to add it manually (might be removed in the future versions).
+injected_fields:
+  DiscoveryConfig:
+    - name: id
+      type: github.com/hashicorp/terraform-plugin-framework/types.StringType
+      computed: true
+      plan_modifiers:
+        - 'github.com/hashicorp/terraform-plugin-framework/tfsdk.UseStateForUnknown()'
+
+# These fields will be excluded
+exclude_fields:
+  # Metadata (we id resources by name on our side)
+  - 'DiscoveryConfig.header.metadata.id'
+  # Read only field
+  - 'DiscoveryConfig.status'
+  # Temporarily exclude access_graph due to Duration field generation issue
+  - 'DiscoveryConfig.spec.access_graph'
+
+# These fields will be marked as Computed: true
+computed_fields:
+  # Metadata
+  - 'DiscoveryConfig.header.metadata.expires'
+  - 'DiscoveryConfig.header.metadata.namespace'
+  - 'DiscoveryConfig.header.metadata.revision'
+  - 'DiscoveryConfig.header.kind'
+  - 'DiscoveryConfig.header.sub_kind'
+
+# These fields will be marked as Required: true
+required_fields:
+  - 'DiscoveryConfig.header.metadata'
```

Comment:
You should add `DiscoveryConfig.header` as a required field as well, since it has required sub fields.